src/networking.c

```diff
@@ -5524,6 +5734,26 @@ void ioThreadReadQueryFromClient(void *data) {
         getKeysFreeResult(&result);
     }
 
+    /* Lookup commands in command queue. */
+    for (int i = c->cmd_queue.off; i < c->cmd_queue.len; i++) {
+        commandParserState *st = &c->cmd_queue.cmds[i];
+        if (!(st->read_flags & READ_FLAGS_PARSING_COMPLETED) || c->argc == 0) continue;
+        struct serverCommand *cmd = lookupCommand(st->argv, st->argc);
+        if (!cmd || !commandCheckArity(c->io_parsed_cmd, c->argc, NULL)) continue;
+        st->cmd = cmd;
+        if (server.cluster_enabled) {
+            /* Offload slot calculation to I/O thread */
+            getKeysResult result;
+            initGetKeysResult(&result);
+            int numkeys = getKeysFromCommand(st->cmd, st->argv, st->argc, &result);
```

Comment:
We should probably write a helper function for this, we don't need to fully populate the key structure, we can just get the first key and compute the slot on that.  Not a blocker here, but would offload the necessary work of finding all the keys when we just need one.